src/lib/markbind/src/patches/htmlparser2.js

```diff
@@ -129,6 +148,189 @@ Tokenizer.prototype._stateText = function(c){
 	}
 };
 
+
+Tokenizer.prototype.specialTagNames = [
+	'script',
+	'style',
+];
+
+/**
+ * Checks whether the token matches one of the first characters of the special tags,
+ * and initialises the _matchingSpecialTagIndexes array with the matches' indexes if so.
+ */
+Tokenizer.prototype.isFirstCharacterSpecialTagCharacter = function(c) {
+	this._matchingSpecialTagIndexes = [];
+	const numSpecialTags = this.specialTagNames.length;
+	const lowerCaseChar = c.toLowerCase();
+	for (let j = 0; j < numSpecialTags; j += 1) {
+		if (lowerCaseChar === this.specialTagNames[j][0]) {
+			this._matchingSpecialTagIndexes.push(j)
+		}
+	}
+
+	if (this._matchingSpecialTagIndexes.length > 0) {
+		this._nextSpecialTagMatchIndex = 1;
```

Comment:
Just wondering - why is this set to `1`? What exactly does the `nextSpecialTagMatchIndex` refer to?

Reply:
It's the next character index to match for the special tags
Here it is `1` since it matched the `0`th character